pyresample/geometry.py

```diff
@@ -449,12 +457,175 @@ class SwathDefinition(CoordinateDefinition):
+        """Compute the oblique mercator projection bouding box parameters."""
+        lines, cols = self.lons.shape
+        lon1, lon, lon2 = np.asanyarray(
+            self.lons[[0, int(lines / 2), -1], int(cols / 2)])
+        lat1, lat, lat2 = np.asanyarray(
+            self.lats[[0, int(lines / 2), -1], int(cols / 2)])
+
+        proj_dict2points = {'proj': 'omerc', 'lat_0': lat, 'ellps': ellipsoid,
+                            'lat_1': lat1, 'lon_1': lon1,
+                            'lat_2': lat2, 'lon_2': lon2}
+
+        lonc, lat0 = Proj(**proj_dict2points)(0, 0, inverse=True)
+        az1, az2, dist = Geod(**proj_dict2points).inv(lonc, lat0, lon1, lat1)
+        return {'proj': 'omerc', 'alpha': float(az1),
+                'lat_0': float(lat0),  'lonc': float(lonc),
+                'no_rot': True, 'ellps': ellipsoid}
+
+    def get_edge_lonlats(self):
+        """Get the concatenated boundary of the current swath."""
+        lons, lats = self.get_boundary_lonlats()
+        blons = np.ma.concatenate([lons.side1, lons.side2,
+                                   lons.side3, lons.side4])
+        blats = np.ma.concatenate([lats.side1, lats.side2,
+                                   lats.side3, lats.side4])
+        return blons, blats
+
+    def compute_bb_proj_params(self, proj_dict):
+        projection = proj_dict['proj']
+        ellipsoid = proj_dict.get('ellps', 'WGS84')
+        if projection == 'omerc':
+            return self._compute_omerc_parameters(ellipsoid)
+        else:
+            raise NotImplementedError('Only omerc supported for now.')
+
+    def compute_optimal_bb_area(self, proj_dict=None):
+        """Compute the "best" bounding box area for this swath with `proj_dict`.
+
+        By default, the projection is Oblique Mercator (`omerc` in proj.4), in
+        which case the right projection angle `alpha` is computed from the
+        swath centerline. For other projections, only the appropriate center of
+        projection and area extents are computed.
+        """
+        if proj_dict is None:
+            proj_dict = {}
+        projection = proj_dict.get('proj', 'omerc')
+        area_id = projection + '_otf'
+        description = 'On-the-fly ' + projection + ' area'
+        lines, cols = self.lons.shape
+        x_size = int(cols * 1.1)
+        y_size = int(lines * 1.1)
+
+        proj_dict = self.compute_bb_proj_params(proj_dict)
+
+        if projection == 'omerc':
+            x_size, y_size = y_size, x_size
+
+        area = DynamicAreaDefinition(area_id, description, proj_dict)
+        lons, lats = self.get_edge_lonlats()
+        return area.freeze((lons, lats), size=(x_size, y_size))
+
+
+class DynamicAreaDefinition():
+    """An AreaDefintion containing just a subset of the needed parameters.
+
+    The purpose of this class is to be able to adapt the area extent and size of
+    the area to a give set of longitudes and latitudes, such that e.g. polar
```

Comment:
"the area to a give set" -> change order of "a" and "give" :-)

Reply:
ack